services/movie.py

```diff
@@ -1,40 +1,46 @@
 from django.db.models import QuerySet
-
+from django.db import transaction
 from db.models import Movie
-
+from typing import List, Optional
 
 def get_movies(
-    genres_ids: list[int] = None,
-    actors_ids: list[int] = None,
+    genres_ids: Optional[List[int]] = None,
+    actors_ids: Optional[List[int]] = None,
+    title: Optional[str] = None
 ) -> QuerySet:
     queryset = Movie.objects.all()
 
     if genres_ids:
-        queryset = queryset.filter(genres__id__in=genres_ids)
+        queryset = queryset.filter(genres__id__in=genres_ids).distinct()
 
     if actors_ids:
-        queryset = queryset.filter(actors__id__in=actors_ids)
-
-    return queryset
+        queryset = queryset.filter(actors__id__in=actors_ids).distinct()
 
+    if title:
+        queryset = queryset.filter(title__icontains=title)
 
-def get_movie_by_id(movie_id: int) -> Movie:
-    return Movie.objects.get(id=movie_id)
+    return queryset
 
+def get_movie_by_id(movie_id: int) -> Optional[Movie]:
+    try:
+        return Movie.objects.get(id=movie_id)
+    except Movie.DoesNotExist:
+        return None
 
 def create_movie(
     movie_title: str,
     movie_description: str,
-    genres_ids: list = None,
-    actors_ids: list = None,
+    genres_ids: Optional[List[int]] = None,
+    actors_ids: Optional[List[int]] = None
 ) -> Movie:
-    movie = Movie.objects.create(
-        title=movie_title,
-        description=movie_description,
-    )
-    if genres_ids:
-        movie.genres.set(genres_ids)
-    if actors_ids:
-        movie.actors.set(actors_ids)
-
-    return movie
+    with transaction.atomic():
+        movie = Movie.objects.create(
```

Comment:
Code Style: It is recommended to use double quotes for strings. Please replace all single quotes with double quotes in your code.